api_app/analyzers_manager/observable_analyzers/greynoiseintel.py

```diff
@@ -0,0 +1,58 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+from greynoise import GreyNoise
+
+from api_app.analyzers_manager import classes
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+
+class GreyNoiseAnalyzer(classes.ObservableAnalyzer):
+    def set_params(self, params):
+        self.api_version = params.get("greynoise_api_version", "v3")
+        self.max_records_to_retrieve = int(params.get("max_records_to_retrieve", 500))
+
+    def run(self):
+
+        if self.api_version == "v2":
+            api_key = self._secrets["api_key_name"]
+            session = GreyNoise(
+                api_key=api_key, integration_name="greynoise-intelowl-v1.0"
+            )
+            context_response = session.ip(self.observable_name)
+            riot_response = session.riot(self.observable_name)
+            response = context_response.copy()
+            response.update(riot_response)
+
+        elif self.api_version == "v3":
+            api_key = self._secrets["api_key_name"]
+            session = GreyNoise(
+                api_key=api_key,
+                integration_name="greynoise-community-intelowl-v1.0",
```

Comment:
how this integration_name is used in the pygreynoise library?

Reply:
This is helpful for us, it is just updating a User-Agent value in the request so we can identify API usage based on integration.  It helps us understand the adoption of the integration and how well it is being used and consumed by our users.

Reply:
thanks for the clarification!